### PR TITLE
Bug fix for generateDataTemplateTask to consume command line option correctly

### PR DESCRIPTION
**Pull request by junchuanwang**
Bugs are introduced in previous attempt https://github.com/linkedin/rest.li/pull/677 refactoring out the DataTemplateGeneratorCmdLineApp

In previous verison, they are always interpreted as "true"

tested by checking out multiproduct, apply the snapshot change and run build

### CODE REVIEW

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/GenerateDataTemplateTask.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
   private boolean _enableArgFile;
   private Boolean _generateLowercasePath;
   private boolean _generateFieldMask;
+  private Boolean _generateImported;
```
Comment (nickibi) [inline gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/GenerateDataTemplateTask.java:51]: What's the use of this new boolean value? I don't see it has been used anywhere.

Comment (nickibi) [reply gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/tasks/GenerateDataTemplateTask.java:51]: Oh, I just saw it.

**Review by nickibi — APPROVED** [review]

**Comment by kshetty-linkedin** [comment]
Looks good to me. Thanks for adding the fix.

File: generator/src/main/java/com/linkedin/pegasus/generator/DataTemplateGeneratorCmdLineApp.java [hunk 2]
```diff
@@ -57,20 +57,20 @@
...
   {
     OPTIONS.addOption("h", "help", false, "Show help.");
-    OptionBuilder.withArgName("Generate imported schemas");
-    OptionBuilder.withLongOpt("generateImported");
+    OptionBuilder.withArgName("Skip generating imported schemas");
+    OptionBuilder.withLongOpt("skipImportedSchemas");
     OptionBuilder.hasArg(false);
-    OptionBuilder.withDescription("Specifies whether to generate classes for externally resolved schemas.");
+    OptionBuilder.withDescription("Specifies whether to skip generating classes for externally resolved schemas.");
     OPTIONS.addOption(OptionBuilder.create('i'));
```
[the remark this answers is not in the document]
Comment (evanw555) [reply generator/src/main/java/com/linkedin/pegasus/generator/DataTemplateGeneratorCmdLineApp.java:64]: +1, "l" doesn't make sense for "case sensitive" anymore.

[the remark this answers is not in the document]
Comment (woodyiorl) [reply generator/src/main/java/com/linkedin/pegasus/generator/DataTemplateGeneratorCmdLineApp.java:64]: @nickibi @evanw555 
Updated to 'c'

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ and what APIs have changed, if applicable.
 
 ## [Unreleased]
 
+## [29.22.5] - 2021-10-06
+- Fix bug for generateDataTemplateTask to consume command line option correctly
```
Comment (evanw555) [inline CHANGELOG.md:18]: Would suggest updating the changelog to be a little more specific about how the command-line args have changed.

**Review by nickibi — APPROVED** [review]

**Pull request merged**